### PR TITLE
if ws is rebinnable, use it, else use its original ws

### PR DESCRIPTION
**Pull request by walshmm**
**Description of work.**
PR addresses [this defect](https://code.ornl.gov/sns-hfir-scse/diffraction/single-crystal/single-crystal-diffraction/-/issues/275)

- [x] `Framework/PythonInterface/plugins/algorithms/ConvertQtoHKLMDHisto.py ` modified
- [x] unit test
- [x] pr against `ornl-next`
- [x] pr against `release-next` 

Companion to PR's [ornl-next [GitHub issue/pull request detail](https://github.com/mantidproject/mantid/pull/31333) , release-next [GitHub issue/pull request detail](https://github.com/mantidproject/mantid/pull/31360)]

**To test:**

Launch Mantid workbench and run the following script from the UI:
```# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np

CreateSimulationWorkspace(Instrument='TOPAZ',
                          BinParams='1,100,10000',
                          UnitX='TOF',
                          OutputWorkspace='cws')
                          
omega, chi, phi = 20, 40, 30

AddSampleLog('cws', "omega", str(omega), "Number")
AddSampleLog('cws', "chi", str(chi), "Number")
AddSampleLog('cws', "phi", str(phi), "Number")

SetGoniometer('cws', Axis0='omega,0,1,0,1', Axis1='chi,0,0,1,1', Axis2='phi,0,1,0,1')

md = ConvertToMD(InputWorkspace='cws', 
                 QDimensions='Q3D',
                 dEAnalysisMode='Elastic', 
                 Q3DFrames='Q_sample',
                 QConversionScales='Q in A^-1',
                 LorentzCorrection='1', 
                 MinValues='-5,-5,-5', 
                 MaxValues='5,5,5')
                 
FakeMDEventData(md, UniformParams='1000')

SetUB(md, u='1,1,0', v='0,0,1', a=5, b=5, c=6, alpha=90, beta=90, gamma=120)

HKL = ConvertQtoHKLMDHisto(md)
HKL *= 2
```
Then right click and open the generated HKL workspace in the slice viewer.  Attempt to zoom and observe no error is thrown as well as it functioning correctly.

#### Release Notes ####

Diffraction Release Notes 6.1
* fixed changes in ConvertQtoHKLMDHisto to accomodate zoom in slice viewer, as a result this algorithm will now drop the connection to the original workspace

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by martyngigg** [comment]
Thanks for looking at this. Can I check we are not reintroducing the bug of #29086?

**Comment by AndreiSavici** [comment]
The new changes look good. Add release notes, or explain why you don't need them

File: Framework/PythonInterface/plugins/algorithms/ConvertQtoHKLMDHisto.py [hunk 1]
```diff
@@ -137,7 +137,7 @@ def PyExec(self):
         SetMDFrame(mdhist, MDFrame='HKL', Axes='0, 1, 2')
 
         self.setProperty("OutputWorkspace", mdhist)
-
+        mdhist.clearOriginalWorkspaces()
```
Comment (martyngigg) [inline Framework/PythonInterface/plugins/algorithms/ConvertQtoHKLMDHisto.py:140]: Agreed that this makes sense 👍🏻

File: Framework/PythonInterface/test/python/plugins/algorithms/ConvertQtoHKLMDHistoTest.py [hunk 2]
```diff
@@ -34,7 +34,7 @@ def test_qtohkl(self):
         hkl = ConvertQtoHKLMDHisto(InputWorkspace=mtd["mde_ws"],
                                    PeaksWorkspace=mtd["peaks"])
         self.assertEqual(hkl.getSpecialCoordinateSystem().name, "HKL")
-
+        self.assertFalse(hkl.hasOriginalWorkspace(0))
```
Comment (martyngigg) [inline Framework/PythonInterface/test/python/plugins/algorithms/ConvertQtoHKLMDHistoTest.py:37]: Nice that we've added a test here 👍🏻

**Review by martyngigg — COMMENTED** [review]
This looks like a good fix. As Andrei says it would be a good to add a line to the framework release notes say that this algorithm now drops its connection to the original workspace.

**Comment by martyngigg** [comment]
I am away for a few days after today so I'm going to drop my assignment of this so you're not waiting for me. Please assume that I am happy with the current changes and have it approved.

**Review by AndreiSavici — APPROVED** [review]
Must be based on `release-next`, otherwise move release notes to v6.2

**Pull request merged**